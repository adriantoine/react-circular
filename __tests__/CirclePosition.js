'use strict';

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _CirclePosition = require('../CirclePosition');

var _CirclePosition2 = _interopRequireDefault(_CirclePosition);

var _enzyme = require('enzyme');

var _enzyme2 = _interopRequireDefault(_enzyme);

var _enzymeAdapterReact = require('enzyme-adapter-react-16');

var _enzymeAdapterReact2 = _interopRequireDefault(_enzymeAdapterReact);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/* eslint-env jest */

_enzyme2.default.configure({ adapter: new _enzymeAdapterReact2.default() });

it('calculates a basic position', function () {
  expect((0, _enzyme.mount)(_react2.default.createElement(
    _CirclePosition2.default,
    { angle: 90, radius: 100 },
    function (_ref) {
      var x = _ref.x,
          y = _ref.y;
      return _react2.default.createElement(
        'text',
        { x: x, y: y },
        'Test'
      );
    }
  ))).toMatchSnapshot();
});

it('calculates a complex position', function () {
  expect((0, _enzyme.mount)(_react2.default.createElement(
    _CirclePosition2.default,
    { angle: 41, radius: 100 },
    function (_ref2) {
      var x = _ref2.x,
          y = _ref2.y;
      return _react2.default.createElement(
        'text',
        { x: x, y: y },
        'Test'
      );
    }
  ))).toMatchSnapshot();
});

it('calculates a complex position with an adjust prop', function () {
  expect((0, _enzyme.mount)(_react2.default.createElement(
    _CirclePosition2.default,
    { angle: 41, radius: 100, adjust: -20 },
    function (_ref3) {
      var x = _ref3.x,
          y = _ref3.y;
      return _react2.default.createElement(
        'text',
        { x: x, y: y },
        'Test'
      );
    }
  ))).toMatchSnapshot();
});

it('uses the render prop when provided', function () {
  expect((0, _enzyme.mount)(_react2.default.createElement(
    _CirclePosition2.default,
    {
      angle: 90,
      radius: 100,
      render: function render(_ref4) {
        var x = _ref4.x,
            y = _ref4.y;
        return _react2.default.createElement(
          'text',
          { x: x, y: y },
          'Rendered'
        );
      }
    },
    function (_ref5) {
      var x = _ref5.x,
          y = _ref5.y;
      return _react2.default.createElement(
        'text',
        { x: x, y: y },
        'NOT RENDERED'
      );
    }
  ))).toMatchSnapshot();
});