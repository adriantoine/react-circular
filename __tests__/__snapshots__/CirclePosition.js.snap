// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calculates a basic position 1`] = `
<CirclePosition
  adjust={0}
  angle={90}
  radius={100}
>
  <text
    x={200}
    y={100}
  >
    Test
  </text>
</CirclePosition>
`;

exports[`calculates a complex position 1`] = `
<CirclePosition
  adjust={0}
  angle={41}
  radius={100}
>
  <text
    x={165.60590289905073}
    y={175.47095802227722}
  >
    Test
  </text>
</CirclePosition>
`;

exports[`calculates a complex position with an adjust prop 1`] = `
<CirclePosition
  adjust={-20}
  angle={41}
  radius={100}
>
  <text
    x={152.48472231924057}
    y={160.37676641782176}
  >
    Test
  </text>
</CirclePosition>
`;

exports[`uses the render prop when provided 1`] = `
<CirclePosition
  adjust={0}
  angle={90}
  radius={100}
  render={[Function]}
>
  <text
    x={200}
    y={100}
  >
    Rendered
  </text>
</CirclePosition>
`;
